DevelopGitHub/GitHubExtension/GitHubExtension.Security.WebApi/FrontEnd/src/client/app/activity/activity.directive.js

```diff
@@ -0,0 +1,42 @@
+(function () {
+    'use strict';
+
+    angular.module('app.activity')
+        .directive('geActivity', activity);
+
+    activity.$inject = [
+        '$templateCache',
+        '$compile',
+        'logger',
+        'ACTIVITY_CONSTANTS',
+        'i18n'
+    ];
+
+    function activity($templateCache, $compile, logger, constants, i18n) {
+        function link(scope, element) {
+            var type = scope.activity.type;
+
+            if (!type) {
+                logger.error(constants.UNKNOWN_TYPE_MESSAGE);
```

Comment:
As I remember if you use logger.error, user will see toastr. So the text should be localized :)